recipes/libtorch/all/conanfile.py

```diff
@@ -0,0 +1,667 @@
+        apply_conandata_patches(self)
+        # Recreate some for add_subdirectory() to work
+        for pkg in ["foxi", "fmt", "FXdiv", "psimd", "mimalloc"]:
+            save(self, os.path.join(self.source_folder, "third_party", pkg, "CMakeLists.txt"), "")
+        # Use FindOpenMP from Conan or CMake
+        modules_dir = os.path.join(self.source_folder, "cmake", "modules")
+        rm(self, "FindOpenMP.cmake", modules_dir)
+
+    def _regenerate_flatbuffers(self):
+        # Re-generate mobile_bytecode_generated.h to allow any flatbuffers version to be used.
+        # As of v24.3.25, only updates the flatbuffers version in the generated file.
+        self.run(f"flatc --cpp --gen-mutable --no-prefix --scoped-enums mobile_bytecode.fbs",
+                 cwd=os.path.join(self.source_folder, "torch", "csrc", "jit", "serialization"))
+
+    def build(self):
+        self._patch_sources()
+        self._pip_install(["pyyaml", "typing-extensions"])
+        if self._depends_on_flatbuffers:
+            self._regenerate_flatbuffers()
+        cmake = CMake(self)
+        cmake.configure()
+        try:
+            cmake.build()
+        except Exception:
+            # The build is likely to run out of memory in the CI, so try again
+            cmake.build(cli_args=["--parallel", "1"])
+
+    def package(self):
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        os.rename(os.path.join(self.package_folder, "share"), os.path.join(self.package_folder, "res"))
+        rmdir(self, os.path.join(self.package_folder, "res", "cmake"))
+        copy(self, "conan-official-libtorch-vars.cmake", self.source_folder, os.path.join(self.package_folder, self._modules_dir))
+
+    @property
+    def _modules_dir(self):
+        return os.path.join("lib", "cmake", "Torch")
+
+    def package_info(self):
+        def _lib_exists(name):
+            return bool(list(Path(self.package_folder, "lib").glob(f"*{name}.*")))
+
+        def _add_whole_archive_lib(component, libname, shared=False):
+            # Reproduces https://github.com/pytorch/pytorch/blob/v2.4.0/cmake/TorchConfig.cmake.in#L27-L43
+            if shared:
+                self.cpp_info.components[component].libs.append(libname)
+            else:
+                lib_folder = os.path.join(self.package_folder, "lib")
+                if is_apple_os(self):
+                    lib_fullpath = os.path.join(lib_folder, f"lib{libname}.a")
+                    whole_archive = f"-Wl,-force_load,{lib_fullpath}"
+                elif is_msvc(self):
+                    lib_fullpath = os.path.join(lib_folder, libname)
+                    whole_archive = f"-WHOLEARCHIVE:{lib_fullpath}"
+                else:
+                    lib_fullpath = os.path.join(lib_folder, f"lib{libname}.a")
+                    whole_archive = f"-Wl,--whole-archive,{lib_fullpath},--no-whole-archive"
```

Comment:
Thanks for your work on this PR--I am not using this library directly but found it through following some github issues on whole archive linking.

For this line--I wonder if we would want to do this with `-Wl,--push-state,--pop-state`? See eg
https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_LINK_LIBRARY_USING_FEATURE.html#loading-a-whole-static-library